### PR TITLE
QCQMC notebook refactor

### PR DESCRIPTION
**Pull request by fdmalone**
Adds some utility functions for interacting with the AFQMC code ipie and tidies up some things.

Depends on #355

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -129,6 +147,43 @@ def fixture_4_qubit_ham_trial_wf_and_blueprint(
     return ham_data, trial_wf_data, bp
 
 
+@pytest.fixture(scope="package")
+def fixture_4_qubit_ham_trial_wf_and_overlap_analysis(
+    fixture_4_qubit_ham_trial_wf_and_blueprint,
+) -> Tuple[HamiltonianData, TrialWavefunctionData, analysis.OverlapAnalysisData]:
```
Comment (dstrain115) [inline conftest.py:153]: Worth a docstring for this prefix?  It's a little bit more complicated and returns a 3-tuple of stuff.

File: docs/qcqmc/index.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# QCQMC
+
+Notebooks outlining how to reproduce the results of [Unbiasing fermionic quantum Monte Carlo with a quantum computer](https://www.nature.com/articles/s41586-021-04351-z).
+Quantum Monte Carlo methods are a class of classical algorithms that can offer
+an efficient solution to the many-electron schroedinger equation but are plagued
+by the `fermion sign problem`. Typically a constraint is introduced to overcome
```
Comment (dstrain115) [inline docs/qcqmc/index.md:6]: nit: I think there should be a comma after Typically.

File: recirq/qcqmc/convert_to_ipie.py [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+        path=overlap_analysis_data.params.base_path.with_suffix(".h5"),
+    )
+
+
+@attrs.frozen(eq=False)
+class OccCoef:
+    occa: np.ndarray
+    occb: np.ndarray
+    coeffs: np.ndarray
+    """A container for the wavefunction bitstrings and coefficients.
```
Comment (dstrain115) [inline recirq/qcqmc/convert_to_ipie.py:177]: Typically, the docstring is placed before the class attributes.

File: recirq/qcqmc/convert_to_ipie.py [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+
+    Args:
+        occa: The spin-alpha (up) determinants. Each entry of occa is an array
+            of integers representing which orbitals are occupied in that
+            determinant.
+        occb: The spin-beta (down) determinants. Each entry of occa is an array
```
Comment (dstrain115) [inline recirq/qcqmc/convert_to_ipie.py:183]: Typo.  should be "Each entry of occb"

File: recirq/qcqmc/fqe_conversion.py [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+    .. math::
+
+        |\\Psi\\rangle &= \\mathrm{(qubit\\ operators)}|0 0\\cdots\\rangle
+        = \\mathrm{(Fermion\\ Operators)}|\\mathrm{vac}\\rangle \\\\
+        |\\Psi\\rangle &= \\sum_iC_i \\mathrm{ops}_{i}|\\mathrm{vac}>
+
+    where the c_{i} are the projection of the wavefunction onto a FCI space.
+
+    Args:
+        cirq_wfn: - coeffcients in the qubit basis.
```
Comment (dstrain115) [inline recirq/qcqmc/fqe_conversion.py:77]: nit: Remove extraneous dash.

File: recirq/qcqmc/fqe_conversion.py [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+    ops: of.QubitOperator, fermion_ind_to_qubit_ind: Optional[Dict[int, int]] = None
+) -> of.FermionOperator:
+    """Converts qubit wavefunction to FQE syntax.
+
+    This takes a qubit wavefunction in the form of a string of qubit operators
+    and returns a string of FermionOperators with the proper formatting for easy
+    digestion by FQE.
+
+    Args:
+        ops (QubitOperator) - a string of qubit operators
```
Comment (dstrain115) [inline recirq/qcqmc/fqe_conversion.py:195]: nit:  Reformat these args to use colons, etc.

**Review by dstrain115 — APPROVED** [review]
Some nits, but seems fine to me.

File: conftest.py (hunk 1 above)
Comment (fdmalone) [reply conftest.py:153]: Done.

File: docs/qcqmc/index.md (hunk 2 above)
Comment (fdmalone) [reply docs/qcqmc/index.md:6]: Done.

File: recirq/qcqmc/convert_to_ipie.py (hunk 3 above)
Comment (fdmalone) [reply recirq/qcqmc/convert_to_ipie.py:177]: Whoops, done.

File: recirq/qcqmc/convert_to_ipie.py (hunk 4 above)
Comment (fdmalone) [reply recirq/qcqmc/convert_to_ipie.py:183]: Done.

File: recirq/qcqmc/fqe_conversion.py (hunk 6 above)
Comment (fdmalone) [reply recirq/qcqmc/fqe_conversion.py:195]: Done.

File: recirq/qcqmc/fqe_conversion.py (hunk 5 above)
Comment (fdmalone) [reply recirq/qcqmc/fqe_conversion.py]: Done.

**Comment by fdmalone** [comment]
Somehow the "previous" cirq version is causing a breakage of the CI and it fails to install numpy (going back to 1.19.x)

**Comment by dstrain115** [comment]
> Somehow the "previous" cirq version is causing a breakage of the CI and it fails to install numpy (going back to 1.19.x)

FYI: this is installing cirq 1.1, which is from 2022 and probably has quite old dependencies.  We probably need to upgrade cirq dependencies in ReCirq.

**Comment by mpharrigan** [comment]
Sorry. The previous, current, next thing was set up before cirq reached 1.0 and there were more version compatibility concerns

**Comment by fdmalone** [comment]
It's weird that I couldn't merge master into this branch on the website following #358, fingers crossed.

**Pull request merged**